scheds/rust/scx_layered/src/main.rs

```diff
@@ -1141,6 +1170,48 @@ fn layer_core_order(
             fastrand::seed(layer_idx.try_into().unwrap());
             fastrand::shuffle(&mut core_order);
         }
+        LayerGrowthAlgo::Topo => {
+            let spec_nodes = spec.nodes();
+            let spec_llcs = spec.llcs();
+            let topo_nodes = topo.nodes();
+
+            if spec_nodes.len() + spec_llcs.len() == 0 {
+                // XXX: fallback to something more sane (round robin when it exists)
+                linear();
+            } else {
+                let mut core_id;
+                for spec_llc in &spec_llcs {
+                    core_id = 0;
+                    for topo_node in topo_nodes {
+                        for core in topo_node.cores().values() {
+                            if core.llc_id != *spec_llc {
+                                core_id += 1;
+                                continue;
+                            }
+                            if !core_order.contains(&core_id) {
```

Comment:
There are a couple of linear searches nested here that make this algorithm pretty expensive, but in the numbers we're looking at I'm sure it's fine.